lib/ts/recipe/dashboard/api/user/create/emailpasswordUser.ts

```diff
@@ -0,0 +1,82 @@
+import { APIInterface, APIOptions } from "../../../types";
+import STError from "../../../../../error";
+import EmailPassword from "../../../../emailpassword";
+import EmailPasswordRecipe from "../../../../emailpassword/recipe";
+import ThridPartyEmailPasswordRecipe from "../../../../thirdpartyemailpassword/recipe";
+import { User } from "../../../../../types";
+import RecipeUserId from "../../../../../recipeUserId";
+import { defaultEmailValidator, defaultPasswordValidator } from "../../../../emailpassword/utils";
+
+type Response =
+    | {
+          status: "OK";
+          user: User;
+          recipeUserId: RecipeUserId;
+      }
+    | {
+          status: "EMAIL_ALREADY_EXISTS_ERROR" | "FEATURE_NOT_ENABLED_ERROR";
+      }
+    | {
+          status: "INPUT_VALIDATION_ERROR";
+          message: string;
+      };
+
+export const createEmailPasswordUser = async (
+    _: APIInterface,
+    tenantId: string,
+    options: APIOptions,
+    __: any
+): Promise<Response> => {
+    try {
+        EmailPasswordRecipe.getInstanceOrThrowError();
+    } catch (error) {
+        try {
+            ThridPartyEmailPasswordRecipe.getInstanceOrThrowError();
+        } catch {
+            return {
+                status: "FEATURE_NOT_ENABLED_ERROR",
+            };
+        }
+    }
+
+    const requestBody = await options.req.getJSONBody();
+
+    const email = requestBody.email;
+    const password = requestBody.password;
+
+    if (email === undefined || typeof email !== "string") {
+        throw new STError({
+            message: "Required parameter 'email' is missing or has an invalid type",
+            type: STError.BAD_INPUT_ERROR,
+        });
+    }
+
+    if (password === undefined || typeof password !== "string") {
+        throw new STError({
+            message: "Required parameter 'password' is missing or has an invalid type",
+            type: STError.BAD_INPUT_ERROR,
+        });
+    }
+
+    const validateEmailError = await defaultEmailValidator(email);
+
+    if (validateEmailError !== undefined) {
+        return {
+            status: "INPUT_VALIDATION_ERROR",
+            message: validateEmailError,
```

Comment:
In the UI, how is this error shown? Cause the status is the same as for invalid password and there is no clear way to tell if it's invalid email or password.

Reply:
added separate statuses for email and password validation errors.